tests/unittests/GraphGradTest.cpp

```diff
@@ -237,3 +237,29 @@ TEST(GraphAutoGrad, checkMatMulToGradTest) {
   EE.compile(CompilationMode::Train, TF);
   EE.compile(CompilationMode::Infer, F);
 }
+
+/// Check that we can differentiate functions that use BatchedReduceAddNode.
+TEST(GraphAutoGrad, checkBatchedReduceAddGradTest) {
+  ExecutionEngine EE;
+  TrainingConfig TC;
+  PlaceholderBindings Bindings;
+
+  // Construct the network:
+  TC.learningRate = 0.001;
+
+  auto &Mod = EE.getModule();
+  Function *F = Mod.createFunction("main");
+
+  TypeRef Ty = Mod.uniqueType(ElemKind::FloatTy, {1, 10});
+  auto *A = Mod.createPlaceholder(ElemKind::FloatTy, {10, 10}, "A", false);
+  auto HandleA = Bindings.allocate(A)->getHandle<float>();
+  HandleA.randomize(-3.0, 3.0, Mod.getPRNG());
+
+  auto *BRA = F->createBatchedReduceAdd("BRA", Ty, A, 0);
```

Comment:
Consider adding a comment to explain that the last argument is the axis:
`auto *BRA = F->createBatchedReduceAdd("BRA", Ty, A, /*axis=*/0);`